src/analysis/retail/deathknight/unholy/modules/Abilities.tsx

```diff
@@ -82,198 +65,230 @@ class Abilities extends CoreAbilities {
+        category: SPELL_CATEGORY.COOLDOWNS,
+        cooldown: 180,
+        gcd: {
+          base: 1500,
         },
       },
       {
-        spell: SPELLS.EMPOWER_RUNE_WEAPON.id,
-        buffSpellId: SPELLS.EMPOWER_RUNE_WEAPON.id,
+        spell: TALENTS.EMPOWER_RUNE_WEAPON_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.EMPOWER_RUNE_WEAPON_TALENT),
         category: SPELL_CATEGORY.COOLDOWNS,
         gcd: null,
         cooldown: 120,
-        castEfficiency: {
-          suggestion: true,
-          recommendedEfficiency: combatant.hasTalent(talents.SUMMON_GARGOYLE_TALENT) ? 0.65 : 0.9,
-          extraSuggestion: (
-            <>
-              You should use this with every <SpellLink spell={talents.SUMMON_GARGOYLE_TALENT} /> if
-              it is talented. Otherwise use it with your other cooldowns when it is available.
-            </>
-          ),
-        },
+        // castEfficiency: {
+        //   suggestion: true,
+        //   recommendedEfficiency: combatant.hasTalent(TALENTS.SUMMON_GARGOYLE_TALENT) ? 0.65 : 0.9,
+        //   extraSuggestion: (
+        //     <>
+        //       You should use this with every <SpellLink spell={TALENTS.SUMMON_GARGOYLE_TALENT} /> if
+        //       it is talented. Otherwise, use it with your other cooldowns when it is available.
+        //     </>
+        //   ),
+        // },
         timelineSortIndex: 1,
-        enabled: combatant.hasTalent(talents.EMPOWER_RUNE_WEAPON_SHARED_TALENT),
       },
-      // defensives
       {
-        spell: talents.SACRIFICIAL_PACT_TALENT.id,
-        category: SPELL_CATEGORY.DEFENSIVE,
+        spell: TALENTS.RAISE_ABOMINATION_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.RAISE_ABOMINATION_TALENT),
+        category: SPELL_CATEGORY.COOLDOWNS,
+        cooldown: 90,
+        gcd: {
+          base: 1500,
+        },
+      },
+      {
+        spell: TALENTS.ABOMINATION_LIMB_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.ABOMINATION_LIMB_TALENT),
+        category: SPELL_CATEGORY.COOLDOWNS,
         cooldown: 120,
         gcd: {
           base: 1500,
         },
       },
+
+      // defensives
       {
-        spell: talents.ICEBOUND_FORTITUDE_TALENT.id,
-        buffSpellId: talents.ICEBOUND_FORTITUDE_TALENT.id,
+        spell: TALENTS.ICEBOUND_FORTITUDE_TALENT.id,
         category: SPELL_CATEGORY.DEFENSIVE,
-        cooldown: 180,
+        cooldown: 120,
         gcd: null,
       },
       {
-        spell: talents.ANTI_MAGIC_ZONE_TALENT.id,
-        buffSpellId: talents.ANTI_MAGIC_ZONE_TALENT.id,
+        spell: SPELLS.ANTI_MAGIC_SHELL.id,
         category: SPELL_CATEGORY.DEFENSIVE,
-        cooldown: 120,
-        gcd: null,
+        cooldown: combatant.hasTalent(TALENTS.ANTI_MAGIC_BARRIER_TALENT)
+          ? combatant.hasTalent(TALENTS.UNYIELDING_WILL_TALENT)
+            ? 60
+            : 40
+          : combatant.hasTalent(TALENTS.UNYIELDING_WILL_TALENT)
+            ? 80
+            : 60,
```

Comment:
this could be written more neatly as:
```suggestion
        cooldown: 80 - Number(combatant.hasTalent(TALENTS.ANTI_MAGIC_BARRIER_TALENT)) * 20 - Number(combatant.hasTalent(TALENTS.UNYIELDING_WILL_TALENT)) * 20,
```